compiler/backend/src/org/jetbrains/kotlin/codegen/inline/InlineCache.kt

```diff
@@ -23,7 +23,7 @@ import org.jetbrains.org.objectweb.asm.commons.Method
 data class MethodId(val ownerInternalName: String, val method: Method)
 
 class InlineCache {
-    val classBytes: SLRUMap<ClassId, ByteArray> = SLRUMap(30, 20)
+    val classBytes: SLRUMap<String, ByteArray> = SLRUMap(30, 20)
```

Comment:
Actually this change would work. But could it cause any weird bug (unclear error messagins) in case of class clashing (that actually should cause a problem in another compiler part):
```
class Foo {
    class Bar {} //Foo$Bar in bytecode
}

and 

class `Foo$Bar` {}
```

Reply:
This cache is only used for loading compiled functions from jar dependencies, so we can assume that the class names are correctly unique, no?

Reply:
>This cache is only used for loading compiled functions from jar dependencies

Good point. Actually such clash will cause runtime problems, but it's not clear how compiler should work here. Now compiler crashes even on non-inline case:

```
// MODULE: lib1
class `Foo$Bar` {
    companion object {
         fun test() = "O"
    }
}

// MODULE: lib2

class Foo {
    class Bar {
        companion object {
             fun test() = "K"
        }
    }
}

// MODULE: main(lib1, lib2)


fun box(): String {
    return `Foo$Bar`.test() +
            Foo.Bar.test()

}
```
```
Caused by: java.lang.AssertionError: Class with incorrect id found: expected /Foo.Bar, actual /Foo$Bar
```